src/marqo/core/structured_vespa_index/structured_vespa_index.py

```diff
@@ -314,17 +314,25 @@ def _get_tensor_search_term(self, marqo_query: MarqoTensorQuery) -> str:
             fields_to_search = self._marqo_index.tensor_field_map.keys()
 
         if marqo_query.ef_search is not None:
-            additional_hits = f', hnsw.exploreAdditionalHits:{marqo_query.ef_search - marqo_query.limit}'
+            target_hits = min(marqo_query.limit + marqo_query.offset, marqo_query.ef_search)
+            additional_hits = max(marqo_query.ef_search - (marqo_query.limit + marqo_query.offset), 0)
         else:
-            additional_hits = ''
+            target_hits = marqo_query.limit + marqo_query.offset
+            additional_hits = 0
 
         terms = []
         for field in fields_to_search:
             tensor_field = self._marqo_index.tensor_field_map[field]
             embedding_field_name = tensor_field.embeddings_field_name
             terms.append(
-                f'({{targetHits:{marqo_query.limit}, approximate:{str(marqo_query.approximate)}{additional_hits}}}'
-                f'nearestNeighbor({embedding_field_name}, {common.QUERY_INPUT_EMBEDDING}))'
+                f'('
+                f'{{'
+                f'targetHits:{target_hits}, '
+                f'approximate:{str(marqo_query.approximate)}, '
+                f'hnsw.exploreAdditionalHits:{additional_hits}'
+                f'}}'
+                f'nearestNeighbor({embedding_field_name}, {common.QUERY_INPUT_EMBEDDING})'
+                f')'
```

Comment:
What about the case where limit = 10, offset = 1000, and efsearch=512
Won't target + additional=512? Don't we need a bigger efsearch to get further into the hit list?

Reply:
Ok I see there is search logic ensuring that efSearch is always >= offset + limit

Reply:
we need to ensure this is clearly specified in documentation

Reply:
Yes if efSearch is too low they'll get empty hits